lib/trie/trie_endtoend_test.go

```diff
@@ -101,19 +101,39 @@ func TestPutAndGetOddKeyLengths(t *testing.T) {
 	runTests(t, trie, tests)
 }
 
-func Fuzz_Trie_PutAndGet(f *testing.F) {
-	trie := NewEmptyTrie()
-	var trieMutex sync.Mutex
-
+func Fuzz_Trie_PutAndGet_Single(f *testing.F) {
 	f.Fuzz(func(t *testing.T, key, value []byte) {
-		trieMutex.Lock()
+		trie := NewEmptyTrie()
 		trie.Put(key, value)
 		retrievedValue := trie.Get(key)
-		trieMutex.Unlock()
 		assert.Equal(t, retrievedValue, value)
 	})
 }
 
+func Test_Trie_PutAndGet_Multiple(t *testing.T) {
+	trie := NewEmptyTrie()
+
+	const numberOfKeyValuePairs = 10000
+
+	generator := newGenerator()
+	keyValues := generateKeyValues(t, generator, numberOfKeyValuePairs)
+	for keyString, value := range keyValues {
+		key := []byte(keyString)
+		trie.Put(key, value)
+
+		// Check value is inserted correctly.
+		retrievedValue := trie.Get(key)
+		assert.Equal(t, retrievedValue, value)
```

Comment:
We might want to stop at the first error and investigate it first.

```suggestion
		require.Equal(t, retrievedValue, value)
```